### PR TITLE
Update the "latest" file after uploading all other artifacts.

### PR DESCRIPTION
**Pull request by ixdy**
Some tools like the SQ expect all artifacts to exist when the latest
file is updated, and this is the behavior that upload-to-gcs.sh used to
use.

Also update the PR directory link at start of build, rather than at end.

Fixes https://github.com/kubernetes/kubernetes/issues/36874, possibly others.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1085)

### CODE REVIEW

File: jenkins/bootstrap.py [hunk 1]
```diff
@@ -327,7 +315,24 @@ def finish(gsutil, paths, success, artifacts, build, version, repo):
...
+    finished_ex = None
+    try:
+        gsutil.upload_json(paths.finished, data)
+    except subprocess.CalledProcessError as finished_ex:
+        logging.warning('Failed to upload %s', paths.finished)
+
+    # Upload the latest build for the job.
+    # Do this last, since other tools expect the rest of the data to be
+    # published when this file is created.
+    for path in {paths.latest, paths.pr_latest}:
```
Comment (fejta) [inline jenkins/bootstrap.py]: I'm not technically sure this is necessary, do we really want to update the latest build if we fail to upload finished.json?

When would we want to do this?

**Comment by ixdy** [comment]
@fejta PTAL

**Comment by spxtr** [comment]
Thanks for fixing this :)

**Pull request merged**